MetaDataset.py

```diff
@@ -886,7 +886,8 @@ def init_seq_order(self, epoch=None, seq_list=None):
     need_reinit = self.epoch is None or self.epoch != epoch
     num_seqs_saved = self._num_seqs
     super(CombinedDataset, self).init_seq_order(epoch=epoch, seq_list=seq_list)  # resets self._num_seqs
-    self.rnd.seed(epoch or 1)
+    full_epoch = (epoch - 1) // self.partition_epoch + 1 if epoch else 1
+    self.rnd.seed(full_epoch)
```

Comment:
This seed was unused and now is only used in ´ _get_random_dataset_seq_order()´. Should we move it there? If not, is it ok to do this partition epoch logic here and assume this is what we want also in other cases?

Reply:
I think we historically had `self.rnd` in `Dataset` but then made it local (to e.g. `get_seq_order_for_epoch`) (I don't quite remember anymore...).
I would always say that local vars are more clean - i.e. having less state in the object. So better move it to `_get_random_dataset_seq_order`.

Btw, this "random_dataset", is this really sth you use? It doesn't sound so useful to me.

Reply:
I've made the random seed local.

I remember having used 'random_dataset' for speech translation to mix audio and text input sequences, where it does not make sense to sort by input length, because you would basically separate the corpora. But anyway, this was the only way to shuffle before I made changes to CombinedDataset so I hope somebody was using it :D